### ISSUE TITLE
fail connect to 17.171.36.30

### ISSUE THREAD

**Issue by 0neday** [tihmstar/tsschecker#74]
[TSSC] opening Buildmanifest for iPhone6,1_10.3.1_14E304
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... success
also requesting APTicket for installType=Update
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... success
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... failure
[TSSR] Request URL set to https://17.171.36.30/TSS/controller?action=2
[TSSR] Sending TSS request attempt 2... failure
[TSSR] Request URL set to https://17.151.36.30/TSS/controller?action=2
[TSSR] Sending TSS request attempt 3... failure
[TSSR] Request URL set to http://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 4... failure
[TSSR] Request URL set to http://17.171.36.30/TSS/controller?action=2

**Comment by OothecaPickle** [comment]
what command did you run?

**Comment by OothecaPickle** [comment]
You should be using the following:

tsschecker -d iPhone6,1 -i 10.3.1 -e <ECID> -s

**Comment by 0neday** [comment]
@OothecaPickle the cmd what I run have no problem, just wonder why have a IP in TSSR Request URL!!

**Comment by 0neday** [comment]
@OothecaPickle, @tihmstar  maybe I need to delete some urls[6] constant variable [https://github.com/tihmstar/tsschecker/blob/master/tsschecker/tss.c] on line 823

**Comment by OothecaPickle** [comment]
Never mind, I'm having this problem as well now. It seems that 17.171.36.30 points to gs-nc.apple.com and 17.151.36.30 points to gs-nwk.apple.com. While I'm not sure what the difference is between these sites and gs.apple.com, replacing these IP addresses with the IP of gs.apple.com(17.111.103.65) does not work. Also, both gs-nwk.apple.com and gs-nc.apple.com seem to be down at the moment, but perhaps they will be back up soon. Until then, I'm not quite sure what to do.

**Comment by OothecaPickle** [comment]
Also, this doesn't seems to matter anyway because blobs are still grabbed.

**Comment by OothecaPickle** [comment]
You just have to let it attempt 15 times.

**Comment by OothecaPickle** [comment]
Never mind, it seems the issue's is actually caused by tsschecker's inability to save noNonce APTickets for the iPhone6,1 (wtf?). Therefore, it just loops through all the urls in the "urls" array and makes 15 attempts to grab noNonce blobs, which, again, for some reason doesn't work when the device is an iPhone6,1. However, both 17.171.36.30 and 17.151.36.30 SHOULD be removed from the array as they are both down at the moment. But, the problem YOU'RE referring to is not technically caused by these IPs being in the array, however these IPs ARE what is causing you to get stuck until curl timeouts. For now, I would simply recommend using [this branch of my fork](https://github.com/OothecaPickle/tsschecker/tree/iphone61), which removes both of the unreachable IP addresses from the array and simply loops through gs.apple.com 15 times (HTTPS and HTTP) when grabbing iPhone6,1 noNonce blobs. This means that for now, you can only grab Erase Install and Update Install APTickets WITH a nonce. Hopefully, @tihmstar will fix this issue soon. :)

**Comment by 0neday** [comment]
@OothecaPickle  I test for iPhone6,1, but can't grab noNonce, just grab restore and update blob

**Comment by OothecaPickle** [comment]
@0neday That's right, that's exactly what I said:

> For now, I would simply recommend using this branch of my fork, which removes both of the unreachable IP addresses from the array and simply attempts to grab noNonce blobs 15 times from gs.apple.com (HTTPS and HTTP) when grabbing iPhone6,1 noNonce blobs. This means that for now, you can only grab Erase Install and Update Install APTickets for the iPhone6,1 WITH a nonce.

**Comment by OothecaPickle** [comment]
You can just use the master branch now: https://github.com/OothecaPickle/tsschecker

**Comment by 0neday** [comment]
is same

**Comment by OothecaPickle** [comment]
@0neday 

> is same

??

**Comment by barberagianluca** [comment]
similar problem here
`tsschecker_macos -d iPhone7,1 -i 10.3.1 --boardconfig n56ap -e XXXXXXXXXXXXX -s --save-path 'iPhone7,1/backup/'` 

> Version: f6e1edae349504c3ec469a64c0309d2f64d217d0 - 230
> [TSSC] manually specified ecid to use, parsed "XXXXXXXXXXXXX" to dec: XXXXXXXXXXXXX hex:xxxxxxxxxxx
> [TSSC] opening firmware.json
> [JSON] counting elements
> [JSON] parsing elements
> [TSSC] got firmwareurl for iOS 10.3.1 build 14E304
> [TSSC] opening Buildmanifest for iPhone7,1_10.3.1_14E304
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 1... success
> also requesting APTicket for installType=Update
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 1... success
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 1... failure
> [TSSR] Request URL set to https://17.111.103.65/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 2... failure
> [TSSR] Request URL set to https://17.111.103.15/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 3... failure
> [TSSR] Request URL set to http://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 4... failure
> [TSSR] Request URL set to http://17.111.103.65/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 5... failure
> [TSSR] Request URL set to http://17.111.103.15/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 6... failure
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 7... failure
> [TSSR] Request URL set to https://17.111.103.65/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 8... failure
> [TSSR] Request URL set to https://17.111.103.15/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 9... failure
> [TSSR] Request URL set to http://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 10... failure
> [TSSR] Request URL set to http://17.111.103.65/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 11... failure
> [TSSR] Request URL set to http://17.111.103.15/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 12... failure
> [TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 13... failure
> [TSSR] Request URL set to https://17.111.103.65/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 14... failure
> [TSSR] Request URL set to https://17.111.103.15/TSS/controller?action=2
> [TSSR] Sending TSS request attempt 15... failure
> Saved shsh blobs!
> iOS 10.3.1 14E304 IS signed!
> 
> iOS 10.3.1 for device iPhone7,1 IS being signed!

**Comment by zweed4u** [comment]
As an owner of a 5S GSM (6,1) I have noticed this as well. Again, I can confirm it is exclusive to certain devices as I also tested with a 7 GSM (9,3) and did not get this issue after the 2-3 successful request attempts. 

I have a few questions as well: 
- Would this infringe upon the success of futurerestore124 on this device? I am using tsschecker232 as the version of this dependency.
- In addition, can we adjust the urls in the array to just exclude these?
